test/phoenix/router/forward_test.exs

```diff
@@ -0,0 +1,130 @@
+defmodule Phoenix.Router.ForwardTest do
+  use ExUnit.Case, async: true
+  use RouterHelper
+
+  defmodule Controller do
+    use Phoenix.Controller
+
+    plug :assign_fwd_conn
+
+    def index(conn, _params), do: text(conn, "admin index")
+    def stats(conn, _params), do: text(conn, "stats")
+    def api_users(conn, _params), do: text(conn, "api users")
+    def api_root(conn, _params), do: text(conn, "api root")
+    defp assign_fwd_conn(conn, _), do: assign(conn, :fwd_conn, conn)
```

Comment:
We need this to test the helpers in the context of a forwarded conn. We store the `fwd_conn` because in the controller context, we have a different path_info and script_name, so the helper will be hitting different code paths.